### ISSUE TITLE
Consider removing dependency on yarn

### ISSUE THREAD

**Issue by bolatovumar** [LN-Zap/zap-desktop#1212]
Yarn was useful back when npm couldn't do deterministic package installs as it does now. Modern npm versions have a `package-lock.json` just as yarn does. Newer npm versions also have the `npm run ci` command for use on the CI server.

The only advantage of yarn over npm at the moment may be some performance boost when you get when installing packages (not sure about this but it's possible).

Having yarn as a dependency just adds an extra package and complexity which may deter new devs from contributing to the project. Fewer dependencies === better imo.

**Comment by mrfelton** [comment]
I don't really see that yarn dependency as an issue since it's pretty widely used and well supported and maintained, and it does provide some benefits (speed in particular). In the past the performance difference was in the order of magnitudes and it was well worth the additional dependency. Maybe the performance of npm installs has improved recently? 

I will say also that I have experienced dependency chain related issues with `package-lock.json`, whilst I have never experienced similar issues with `yarn.lock`.

Another feature of yarn that I do like is the workspaces support, and whilst we aren't using it yet on this project, we are using it on some of our other projects and I'd like to keep our devops processes consistent across our projects if possible.

That said, I'm open to this change - I'm all for using stanzdardized and vanilla approaches if they make sense - though don't see this change as a particular high priority at this point. I would guess that most developers that are able to contribute to this project would be capable of installing yarn without any issues.

**Comment by bolatovumar** [comment]
Yeah,. I don't think it's a particularly high priority item and don't
really see an issue with using yarn. I generally prefer using more vanilla
tooling unless absolutely necessary. We used yarn for s short while at my
workplace but switched back to nom since it improved quite a bit.

See this blog post about the npm ci command:
https://blog.npmjs.org/post/171556855892/introducing-npm-ci-for-faster-more-reliable

They claim it improves the install performance by 2x in some cases in
comparison to regular nom install and even yarn.

I guess I can do some testing to see which two of the options is faster?

It's really up to you whether you think this would be worthy to change.

**Comment by bolatovumar** [comment]
@mrfelton another advantage if the newer npm versions I forgot to mention is that they provide a vulnerability report on each install. For example, it found one high severity vulnerability somewhere in the dependency chain. Obviously it can only report known vulnerabilities but it's still a lot better than nothing.

**Comment by bolatovumar** [comment]
Alright, I tested yarn vs npm.

I'm using the latest versions of npm and yarn (6.5.0 and 1.12.3 respectively) with NodeJS v10.15.0 on a MacBook Air (the older version) with ~200mbps down connection. I ran all install commands with yarn and npm cache cleared right before the install command. I also deleted the node_modules folder prior to each new install.

Here are the results:
npm ci: **127.875s**
npm i: **148s**
yarn: **184.41s**

Looks like npm is the clear winner here.

**Comment by mrfelton** [comment]
This is encouraging. Shaving time off our CI builds is quite a compelling prospect. If you want to get a patch together I'd be happy to review.

**Comment by bolatovumar** [comment]
I will do some additional testing on a more powerful machine and a lest powerful one with different connection speeds to see if the results replicate. If they do, I would say the switch would be definitely worth it.

**Comment by bolatovumar** [comment]
@mrfelton finally got around to testing the initial package install on another (slower) machine. Yarn is v1.13.0 and NPM is v6.6.0.

yarn: **244s**
npm i: **214s**
npm ci: **142s**

Looks like npm has the edge.
